OSSpecific/src/artofillusion/osspecific/DesktopAdapterJDK8.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package artofillusion.osspecific;
+
+
+import com.apple.eawt.AboutHandler;
+import com.apple.eawt.AppEvent;
+import com.apple.eawt.Application;
+import com.apple.eawt.OpenFilesHandler;
+import com.apple.eawt.PreferencesHandler;
+import com.apple.eawt.QuitHandler;
+import com.apple.eawt.QuitResponse;
+
+
+/**
+ *
+ * @author maksim.khramov
+ */
+public class DesktopAdapterJDK8 extends DesktopAdapter implements AboutHandler, OpenFilesHandler, PreferencesHandler, QuitHandler {
```

Comment:
Line 22 can't work. The `implements` clause will throw `ClassNotFound` X4 if we try to load on JDK 8 - the interface classfiles need to be present to load correctly.

Reply:
Right. And I catching this on plugin onApplicationStarting code to fallback to new JDK9 implementation...

Reply:
How? These interfaces were added as part of Java 9... Even marked so in the source.

Reply:
No. Apple specific com.apple.eawt.* classes available in JDK8 on Mac platform and dropped and replaced with java.awt.Desktop in JDK9. The new JDK9 interfaces from java.awt.* in DesktopAdapter9 has same names but other method parameters.
The OE library allows us to see and use this com.apple.eawt.* on other platforms and/or JDK9+

Reply:
Ahh...